pkg/controllers/jobset_controller.go

```diff
@@ -358,7 +358,7 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 		}
 
 		// If pod DNS hostnames are enabled, create a headless service per replicatedjob.
-		if dnsHostnamesEnabled(&rjob) {
+		if dnsHostnamesEnabled(js) {
```

Comment:
we should move this outside the loop.

Reply:
Good call! This required removing the replicated job as an argument (which was added as a selector) so since we don't need that anymore (the jobset name is sufficient) I could remove it and put it outside of the loop. I originally kept it because I wasn't sure if the extra selector label could be useful.